contributor-docs/adrs/adr-005.md

```diff
@@ -0,0 +1,65 @@
+# ADR 5: Migration plan for deprecated components
+
+## Status
+
+Proposed
+
+## Context
+
+As we work on maturity of our components, we will sometimes need to deprecate components that are replaced with new components.
+
+## Here are the 3 proposed stages:
+
+### Stage 1
+
+Start new component outside the main bundle. Folks who want to try it, need to explicitly import it from the `drafts` bundle.
+
+`import { ActionMenu} from '@primer/react/drafts'`
+
+The contract with consumers is - you are opting into a rewrite of the old component that might not cover all the cases of the old component yet. If you are using both the old and new version of the component in different pages, you are paying some additional bundlesize cost.
+
+Note: If it's a 1:1 replacement, it's useful to keep the same name.
+
+### Stage 2
+
+After we have the confidence that this component is better than the old version of it, we swap the components and move the old component out of the main bundle.
+
+This is a breaking change! It’s recommended that you start using the new component, but if you’d like to push that effort to the future, we give you an easy way out -
+
+`import { ActionMenu } from '@primer/react/deprecated'`
+
+The deprecated component does not accept new features requests.
+
+Reason: Because we have a single bundle for all components, you can not pick the components you want to upgrade. This can results in additional unrelated work.
+
+### Stage 3
+
+After 3 months of living in the `deprecated` bundle, a component is retired/deleted from the codebase.
```

Comment:
`3 months` is an example, do you feel this should have a longer timeline?

Reply:
I think 3 months is fine for most cases. We can always make an exception if we need to.